app/models/qc_result_factory.rb

```diff
@@ -86,18 +90,22 @@ def save
     end
 
     def working_dilution?
-      assay_type == 'Working dilution'
+      return false if plate.blank?
+      plate.plate_purpose.name == 'Working Dilution'
```

Comment:
WorkingDilution plates actually have a different class, so this could be used to avoid the need for a magic string. (Which would also allow you just to use the :dilution_plate_purpose factory in the specs)